apps/android/app/src/main/java/chat/simplex/app/views/chat/SendMsgView.kt

```diff
@@ -150,10 +86,216 @@ fun SendMsgView(
+            rec.stop(recordingInProgress)
+            recordingTimeRange = recordingTimeRange.first..System.currentTimeMillis()
+            filePath.value?.let { onAudioAdded(it, (recordingTimeRange.last - recordingTimeRange.first).toInt(), true) }
+          }
+          val startStopRecording = {
+            when {
+              !permissionsState.allPermissionsGranted -> permissionsState.launchMultiplePermissionRequest()
+              recordingInProgress.value -> stopRecordingAndAddAudio()
+              filePath.value == null -> {
+                showRecordingUi(true)
+                recordingTimeRange = System.currentTimeMillis()..0L
+                filePath.value = rec.start(recordingInProgress, stopRecordingAndAddAudio)
+              }
+            }
+          }
+          var stopRecOnNextClick by remember { mutableStateOf(false) }
+          val context = LocalContext.current
+          DisposableEffect(stopRecOnNextClick) {
+            val activity = context as? Activity ?: return@DisposableEffect onDispose {}
+            if (stopRecOnNextClick) {
+              // Lock orientation to current orientation because screen rotation will break the recording
+              activity.requestedOrientation = if (activity.resources.configuration.orientation == Configuration.ORIENTATION_PORTRAIT)
+                ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+              else
+                ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+            }
+            // Unlock orientation
+            onDispose { activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED }
+          }
+          val cleanUp = { remove: Boolean ->
+            if (remove) filePath.value?.let { File(it).delete() }
+            filePath.value = null
+            recordingInProgress.value = false
+            stopRecOnNextClick = false
+            recordingTimeRange = 0L..0L
+            showRecordingUi(false)
+          }
+          LaunchedEffect(cs.preview) {
+            if (cs.preview !is ComposePreview.VoicePreview && filePath.value != null) {
+              // Pressed on X icon in preview
+              cleanUp(true)
+            }
+          }
+          val interactionSource = interactionSourceWithTapDetection(
+            onPress = {
+              if (filePath.value == null) startStopRecording()
+            },
+            onClick = {
+              if (!recordingInProgress.value && filePath.value != null) {
+                sendMessage()
+                cleanUp(false)
+              } else if (stopRecOnNextClick) {
+                stopRecordingAndAddAudio()
+                stopRecOnNextClick = false
+              } else {
+                // tapped and didn't hold a finger
+                stopRecOnNextClick = true
+              }
+            },
+            onCancel = startStopRecording,
+            onRelease = startStopRecording
+          )
+          Spacer(Modifier.weight(1f))
+          Icon(
```

Comment:
It seems it misses `.clip(CircleShape)` and the color should depend as well then for when it's "ready for sent" (it should be white arrow upward in accent circle).

Also these set of buttons also should take into account `cs.sendEnabled()` - both record and send, and the background color if send is disabled is HighOrLowlight instead of primary - see `if (!showVoiceButton)`